cts-java/src/test/java/org/ga4gh/cts/api/references/ReferenceSetsSearchIT.java

```diff
@@ -21,49 +21,132 @@
+        assertThat(refSets).isNotEmpty();
+        refSets.stream().forEach(rs -> assertThat(rs.getAssemblyId()).isEqualTo(assemblyId));
+    }
 
-        refSets.stream().forEach(rs -> assertThat(rs.getNcbiTaxonId()).isEqualTo(ncbiTaxonId));
+    /**
+     * Search for reference set by md5 checksum.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void searchForReferenceSetByMD5Checksum() throws AvroRemoteException {
+        final String md5checksum = "90977a37195d3fd247e4916b5b4cbae8";
+        final String assemblyId = "hg37";
+
+        final SearchReferenceSetsRequest req = 
+            SearchReferenceSetsRequest.newBuilder()
+                                        .setMd5checksums(aSingle(md5checksum))
+                                        .build();
+        final SearchReferenceSetsResponse resp = client.references.searchReferenceSets(req);
+        
+        final List<ReferenceSet> refSets = resp.getReferenceSets();
+        assertThat(refSets).isNotEmpty();
         refSets.stream().forEach(rs -> assertThat(rs.getAssemblyId()).isEqualTo(assemblyId));
+        refSets.stream().forEach(rs -> assertThat(rs.getMd5checksum()).isEqualTo(md5checksum));
+}
 
-        // do query 2 and test 2
-        for (ReferenceSet refSet : refSets) {
-            final String id = refSet.getId();
-            final ReferenceSet fetchedRefSet = client.references.getReferenceSet(id);
-            assertThat(fetchedRefSet.getId()).isEqualTo(id);
-        }
+    /**
+     * Search for reference set by assemblyId.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void searchForReferenceSetByAssemblyId() throws AvroRemoteException {
+        final String assemblyId = "hg37";
+        final String md5checksum = "90977a37195d3fd247e4916b5b4cbae8";
+
+        final SearchReferenceSetsRequest req = 
+            SearchReferenceSetsRequest.newBuilder()
+                                        .setAssemblyId(assemblyId)
+                                        .build();
+        final SearchReferenceSetsResponse resp = client.references.searchReferenceSets(req);
+        
+        final List<ReferenceSet> refSets = resp.getReferenceSets();
+        assertThat(refSets).isNotEmpty();
+        refSets.stream().forEach(rs -> assertThat(rs.getAssemblyId()).isEqualTo(assemblyId));
+        refSets.stream().forEach(rs -> assertThat(rs.getMd5checksum()).isEqualTo(md5checksum));
     }
 
+    /**
+     * Verify that we're looking up the right species (NCBI Taxonomy ID).
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkReferenceSetHasCorrectTaxonId() throws AvroRemoteException {
+        final String accession = "GA4GH_CTS_01";
```

Comment:
I thought I had seen it mentioned `accessionId`, but you're right, that doesn't occur anywhere in the source code!  This comment is really about consistency, so if you call it `accession` throughout, I'm happy.